File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-6.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        Deque<TreeNode> queue = new ArrayDeque<TreeNode>();
        queue.add(root);
        
        while (!queue.isEmpty()) {
            int levelLength = queue.size();  // number of elements in the level
            List<Integer> sub = new ArrayList<>();
            for (int i = 0; i < levelLength; i++) {
                TreeNode curr = queue.removeFirst();  // popping then adding children (BOTH LEFT AND RIGHT)
                // it's being popped from queue but it's still part of Tree structure
                if (curr.left != null) {
                    queue.addLast(curr.left);
                    sub.add(curr.left.val);
                }
                if (curr.right != null) {
                    queue.addLast(curr.right);
                    sub.add(curr.right.val);
                }
            }
            res.add(sub);
            level++;
        }
        return res;
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root == None: # empty list, with no Tree Nodes
            return 0
        from collections import deque
        queue = deque()
        queue.append((root, 0))
        size = self.helper(root, queue)
        return size
    
    def helper(self, node: Optional[TreeNode], queue) -> int:
        while len(queue) > 0:
            curr, size = queue.popleft()
            if curr != None:
                size += 1
                queue.append((curr.left, size))
                queue.append((curr.right, size))
        return size